nixos/modules/services/networking/vpp.nix

```diff
@@ -0,0 +1,543 @@
+      };
+  };
+
+  vppOpts =
+    { name, ... }:
+    {
+      options = {
+        enable = mkEnableOption "FD.io's Vector Packet Processor, a high-performance userspace network stack.";
+
+        name = mkOption {
+          type = types.str;
+          default = name;
+          description = ''
+            Name is used as a suffix for the service name.
+            By default it takes the value you use for `<instance>` in:
+            {option}`services.vpp.instances.<instance>`
+          '';
+        };
+
+        package = mkPackageOption pkgs "vpp" { };
+
+        kernelModule = mkOption {
+          type = types.nullOr types.str;
+          default = "uio_pci_generic";
+          example = "vfio-pci";
+          description = "UIO kernel driver module to load before starting this VPP instance. Set to `null` to disable this functionality.";
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = "vpp";
+          example = "vpp-main";
+          description = "Group that grants users in it privileges to control this instance via {command}`vppctl`.";
+        };
+
+        settings = mkOption {
+          description = ''
+            Configuration for VPP, see <https://fd.io/docs/vpp/master/configuration/reference.html> for details.
+
+            Nix value declared here will be translated directly to the config format VPP uses. Attributes
+            called `extraConfig` will be inserted verbatim into the resulting config file.
+
+            Top-level attrsets, lists and primitives get parsed as expected, i.e. `foo = { bar = true; };`
+            becomes `foo { bar }` in the final config file. Nested attrsets on the other hand, such as
+            {option}`dpdk.dev`, get parsed differently. For example, this config:
+            ```nix
+            settings = {
+              dpdk = {
+                dev = {
+                  "foo".name = "bar";
+                  "baz".name = "qux";
+                };
+              };
+            };
+            ```
+            Would get parsed into the following:
+            ```
+            dpdk {
+              dev foo { name bar }
+              dev baz { name qux }
+            }
+            ```
+            Notice how `dev.foo` becomes `dev foo { }` instead of `dev { foo { } }`.
+          '';
+          defaultText = ''
+            unix = {
+              nodaemon = true;
+              nosyslog = true;
+              cli-listen = "/run/vpp/\${name}-cli.sock";
+              gid = config.services.vpp.instances.\${name}.group;
+              startup-config = config.services.vpp.instances.\${name}.startupConfigFile;
+              cli-prompt = "vpp-\${name}";
+            };
+            api-segment = {
+              prefix = "\${name}";
+              gid = config.services.vpp.instances.\${name}.group;
+            };
+          '';
+          example = lib.literalExpression ''
```

Comment:
We should try to condense this, too.

Reply:
Removed a bunch of unnecessary options, is this short enough? I'd prefer keeping the comment in there since I think it's pretty useful.